R/combinevariablesetsasbinary.R

```diff
@@ -151,8 +161,45 @@ questionToBinary <- function(x) {
         binary.version <- AsNumeric(x, binary = TRUE, name = levels(x))
         colnames(binary.version) <- levels(x) 
         binary.version[is.na(x), ] <- NA
+        attr(binary.version, "originalquestiontype") <- "Pick One"
         return(binary.version)
     }
 
     stop("Unsupported data type: ", question.type)
 }
+
+# Function to expand the number of columns in the binary data
+# when there are fewer columns than expected. expected.columns
+# should be a vector of column names. 
+fillInCategoriesWhenNotPresent <- function(binary.data, expected.columns, pick.any.all.missing = TRUE) {
+    
+    current.colnames <- colnames(binary.data)
+
+    if (all(expected.columns %in% colnames(current.colnames)))
+        return(binary.data)
+
+    new.colnames <- expected.columns[! expected.columns %in% current.colnames]
```

Comment:
`colnames` is called twice by accident I believe. i.e. the RHS of `%in%` is `colnames(colnames(binary.data))`
could replace this with a call to `setdiff` to find the new ones.
```suggestion
   current.colnames <- colnames(binary.data)
   new.colnames <- setdiff(expected.column, current.colnames)
    if (length(new.colnames) == 0L)
        return(binary.data)
```